tcs/csp_solver_mono_eq_methods.cpp

```diff
@@ -28,1882 +28,960 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+                    xy2.x = t_ts_solved;   //[s]
+                    xy2.y = diff_t_ts_calculated;
+
+                    double tol_solved = std::numeric_limits<double>::quiet_NaN();  //[s]
+                    int iter_solved = -1;
+                    int t_ts_code = 0;
+                    try
+                    {
+                        // Want solver to miss such that solved startup time is greater than components require
+                        // If components require slightly more, then they won't completely finish startup and next time period will require
+                        //       a very short timestep of startup
+                        t_ts_code = c_T_cold_solver.solve(xy1, xy2, -tol_t_ts_target, t_ts_solved, tol_solved, iter_solved);
+                    }
+                    catch (C_csp_exception)
+                    {
+                        *target = std::numeric_limits<double>::quiet_NaN();
+
+                        mpc_csp_solver->reset_time(m_t_ts_initial);
+
+                        return -4;
+                    }
+                    if (t_ts_code != C_monotonic_eq_solver::CONVERGED)
+                    {
+                        if (t_ts_code > C_monotonic_eq_solver::CONVERGED && fabs(tol_solved) < 0.1*m_t_ts_initial)
+                        {
+                            std::string msg = util::format("At time = %lg power cycle startup time iteration "
+                                " only reached a convergence"
+                                "= %lg [s]. Check that results at this timestep are not unreasonably biasing total simulation results",
+                                mpc_csp_solver->mc_kernel.mc_sim_info.ms_ts.m_time / 3600.0, tol_solved);
+                            mpc_csp_solver->mc_csp_messages.add_message(C_csp_messages::NOTICE, msg);
+                        }
+                        else
+                        {
+                            *target = std::numeric_limits<double>::quiet_NaN();
+                            
+                            mpc_csp_solver->reset_time(m_t_ts_initial);
+
+                            return -5;
+                        }
+                    }
+                }
+            }
+            else
+            {
+                t_ts_solved = t_ts_max_local;
+            }
+        }
+        else
+        {
+            return -123;
+        }
+    }
+
+
+    // Have been mucking with mc_kernel.mc_sim_info.ms_ts.m_time, so need to reset these
+    mpc_csp_solver->mc_kernel.mc_sim_info.ms_ts.m_step = t_ts_solved;						//[s]
+    mpc_csp_solver->mc_kernel.mc_sim_info.ms_ts.m_time = mpc_csp_solver->mc_kernel.mc_sim_info.ms_ts.m_time_start + t_ts_solved;		//[s]
+
+    *target = calc_meq_target();
+
+    return 0;
 }
 
-int C_csp_solver::C_mono_eq_pc_target_tes_dc__T_cold::operator()(double T_htf_cold /*C*/, double *diff_T_htf_cold /*-*/)
+int C_csp_solver::C_MEQ__timestep::operator()(double t_ts_guess /*s*/, double *target /*varying*/)
```

Comment:
rename "t_ts_guess".   time+timestep+guess?     if so, maybe to "timestep_guess"

Reply:
current name is ok